modules/diskq/diskq.c

```diff
@@ -30,6 +30,16 @@
 #include "logqueue-disk-non-reliable.h"
 #include "persist-state.h"
 
+static gboolean
+log_queue_disk_is_file_in_directory(const gchar *file, const gchar *directory)
+{
+  gchar *basedir = g_path_get_dirname(file);
+  const gboolean result = (0 == strcmp(basedir, directory));
```

Comment:
An `strcmp` call might not be enough here, because `disk_queue_options_set_dir()` uses the specified directory string from the configuration file without any normalization.

For example, the directory specified by the user can be:
`"/var/disk-buffers/"`
or just
`"/var/disk-buffers"`

Reply:
I've updated, to normalize during the configuration time. The only side effect of this solution, if a `syslog-ng` is started with a persist file, that already has a not normalized path. But in this case, we only do not print this warning, and the functionality is not changed.
Does it acceptable ?

Reply:
IMHO, it is perfectly acceptable.